core/src/main/java/com/linecorp/armeria/server/Http1RequestDecoder.java

```diff
@@ -409,8 +409,7 @@ public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exc
             // HTTP/1 doesn't support draining that signals clients about connection shutdown but still
             // accepts in flight requests. Simply destroy KeepAliveHandler which causes next response
             // to have a "Connection: close" header and connection to be closed after the next response.
-            destroyKeepAliveHandler();
```

Comment:
super nit; I feel like we can just remove this method altogether in this class

Reply:
Sorry, I didn't get the intention. Could you elaborate on it?

Reply:
I thought we don't need to call the method anymore since `KeepAliveHandler` will remove itself on channel disconnection

Reply:
Good point. Besides disconnection, KeepAliveHandler for HTTP/1 is destroyed while upgrading. The case is converted by `encoder.close()` that closes the underlying `KeepAliaveHandler` as well.
https://github.com/line/armeria/blob/c6f7b8ee907b3b5626cdf29319eb5a70dff7b55b/core/src/main/java/com/linecorp/armeria/server/Http1RequestDecoder.java#L368

The code related to keep-alive is quite messy. Let me send a cleanup PR after the LINE OSS sprint. 😄